Source/Details/ASCollectionViewFlowLayoutCalculator.m

```diff
@@ -0,0 +1,35 @@
+//
+//  ASCollectionViewFlowLayoutCalculator.m
+//  AsyncDisplayKit
+//
+//  Created by Huy Nguyen on 28/2/17.
+//  Copyright © 2017 Facebook. All rights reserved.
+//
+
+#import <AsyncDisplayKit/ASCollectionViewFlowLayoutCalculator.h>
+#import <AsyncDisplayKit/ASCellNode.h>
+#import <AsyncDisplayKit/ASCollectionElement.h>
+#import <AsyncDisplayKit/ASElementMap.h>
+#import <AsyncDisplayKit/ASStackLayoutSpec.h>
+
+@implementation ASCollectionViewFlowLayoutCalculator
```

Comment:
This class is so impressively elegant.  It's a pure expression of what the coming together of the entire core team's work has produced.